src/main/org/apache/tools/ant/taskdefs/optional/ssh/SSHBase.java

```diff
@@ -235,6 +259,37 @@ public void init() throws BuildException {
         this.port = SSH_PORT;
     }
 
+    /**
+     * Load the SSH configuration file.
+     * @throws BuildException on error
+     */
+    protected void loadSshConfig() throws BuildException {
+        if (sshConfig != null) {
+            if (!new File(sshConfig).exists()) {
+                throw new BuildException("The SSH configuration file specified doesn't exist: " + sshConfig);
+            }
+            
+            log("Loading SSH configuration file " + sshConfig, Project.MSG_DEBUG);
+            ConfigRepository.Config config = null;
+            try {
+                config = OpenSSHConfig.parseFile(sshConfig).getConfig(host);
+            } catch (IOException e) {
+                throw new BuildException("Failed to load the SSH configuration file " + sshConfig, e);
+            }
+            
+            host = config.getHostname();
+            
+            if (userInfo.getName() == null) {
```

Comment:
Should this check perhaps happen before even trying to load the `sshConfig` file? So as to avoid the whole loading and parsing of the file?

Perhaps load the file only if userInfo doesn't have the name or keyfile available?